sdk/Lusid.Sdk.Tests/Utilities/ApiRetryHandlerTests.cs

```diff
@@ -0,0 +1,107 @@
+        private readonly Policy<IRestResponse> _initialRetryPolicy = RetryConfiguration.RetryPolicy;
+        private HttpListener _httpListener;
+        private const string ListenerUriPrefix = "http://localhost:4444/";
+        
+        [SetUp]
+        public void SetUp()
+        {
+            _httpListener = new HttpListener();
+            _httpListener.Prefixes.Add(ListenerUriPrefix);
+            
+            var testApiConfig = TestLusidApiFactoryBuilder.CreateApiConfiguration(null);
+            testApiConfig.ApiUrl = ListenerUriPrefix;
+            
+            _apiFactory = new LusidApiFactory(testApiConfig);
+        }
+
+        [Test]
+        [Explicit("This passes locally, but not in the pipeline, and no idea why. Setting to explicit until resolved.")]
+        public void CallGetPortfoliosApi_WhenPollyRetryPolicyConfigured_PollyIsTriggered()
+        {
+            const int expectedNumberOfRetries = ApiRetryHandler.MaxRetryAttempts;
+            _httpListener.Start();
+            for (var i = 0; i < expectedNumberOfRetries + 1; i++)
+            {
+                 _httpListener.BeginGetContext(result =>
+                {
+                    var listener = (HttpListener) result.AsyncState;
+                    // Call EndGetContext to complete the asynchronous operation.
+                    var context = listener.EndGetContext(result);
+
+                    // Obtain a response object.
+                    var response = context.Response;
+
+                    // Abort the response. This returns 0 status code.
+                    response.Abort();
+                }, _httpListener);
+            }
+            
+            var currentRetry = 0;
+            var testRetryPolicy = Policy
+                .HandleResult<IRestResponse>(restResponse => restResponse.StatusCode == 0)
+                .Retry(
+                    retryCount: expectedNumberOfRetries,
+                    (result, i) =>
+                    {
+                        Console.WriteLine($"If you see this, then polly retry works. Retry number: {i}");
+                        currentRetry = i;
+                    });
+
+            RetryConfiguration.RetryPolicy = testRetryPolicy;
+
+            // Calling GetPortfolio or any other API triggers the flow that triggers polly
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any");
+            
+            Assert.That(currentRetry, Is.EqualTo(expectedNumberOfRetries));
+            // In the future 0 error codes with throw an error after retries exceeded
+            Assert.That(sdkResponse, Is.Null);
+        }
+
+        [Test]
+        public void CreateLusidFactory_WhenRetryPolicyIsNull_AssignsDefaultRetryPolicy()
+        {
+
+            RetryConfiguration.RetryPolicy = null;
+            
+            var newFactory = new LusidApiFactory(new Configuration());
+
+            Assert.That(RetryConfiguration.RetryPolicy, Is.Not.Null);
```

Comment:
This is assigning the ApiRetryHandler right? Can you assert on that instead of just that it is not null.

Reply:
Again, is this the behaviour that we want where we modify the global RetryConfiguration? Or should we just update the config for the ApiClient's produced by the instance of the ApiClientFactory.

Reply:
yeah will have to figure out how to compare policies

Reply:
That's okay let's stick with global. Looks to be difficult to modify this without modifying the ApiClient itself. You can do `Assert.That(RetryConfiguration.RetryPolicy, Is.EqualTo(PollyApiRetryHandler.InternalExceptionRetryPolicyWithFallback));`

Reply:
Done

Reply:
Resolved, nice!